pkg/operator/applier/kubectl.go

```diff
@@ -132,6 +136,138 @@ commonAnnotations:
 	return stdout, stderr, errors.Wrap(err, "failed to run kubectl apply")
 }
 
+// ApplyCreateOrPatch attempts to run a `kubectl apply` on the yaml document. If it fails
+func (c *Kubectl) ApplyCreateOrPatch(targetNamespace string, slug string, yamlDoc []byte, dryRun bool, wait bool, annotateSlug bool) ([]byte, []byte, error) {
+
+	stdout, stderr, err := c.Apply(targetNamespace, slug, yamlDoc, dryRun, wait, annotateSlug)
+	if !strings.Contains(string(stderr), "metadata.annotations: Too long") || err == nil {
```

Comment:
this would be more readable if you checked error == nil separate from handling the special error case